addons/sourcemod/scripting/vsh/bosses/boss_announcer.sp

```diff
@@ -428,9 +517,80 @@ public Action Timer_AnnouncerChangeTeam(Handle hTimer, int iClient)
 		}
 	}
 	
+	//Allow sentries to target this fella from now on
+	SetEntityFlags(iClient, (GetEntityFlags(iClient) & ~FL_NOTARGET));
+	
 	//Dont give health overheal
 	SetEntProp(iClient, Prop_Send, "m_iHealth", SDK_GetMaxHealth(iClient));
 	
 	//TF2_AddCondition(iClient, TFCond_Buffed, TFCondDuration_Infinite);
 	TF2_AddCondition(iClient, TFCond_DefenseBuffed, TFCondDuration_Infinite);
+}
+
+public void Announcer_SetBuildingTeam(int iBuilding, TFTeam nTeam, int iNewBuilder = -1)
+{
+	int iBuilder = GetEntPropEnt(iBuilding, Prop_Send, "m_hBuilder");
+	
+	//Remove the building from the original builder so it doesn't explode on team switch
+	SDK_RemoveObject(iBuilder, iBuilding);
+	
+	int iTeam = view_as<int>(nTeam);
+	SetVariantInt(iTeam);
+	AcceptEntityInput(iBuilding, "SetTeam");
+	
+	//You can set the server as the builder if you want, I'm not stopping you
+	if (iNewBuilder > -1)
+		SetEntPropEnt(iBuilding, Prop_Send, "m_hBuilder", iNewBuilder);
+	
+	//Mini-sentries use different skins, adjust accordingly
+	if (GetEntProp(iBuilding, Prop_Send, "m_bMiniBuilding"))
+		SetEntProp(iBuilding, Prop_Send, "m_nSkin", iTeam);
+	else
+		SetEntProp(iBuilding, Prop_Send, "m_nSkin", iTeam-2);
+	
+	char sClassname[32];
+	GetEntityClassname(iBuilding, sClassname, sizeof(sClassname));
+	
+	//Disable teleporters for a little bit to reset the effects' colors
+	if (StrEqual(sClassname, "obj_teleporter"))
+	{
+		TF2_StunBuilding(iBuilding, 0.1);
+	}
+	
+	//Actually just disable the dispenser's screen, should check up on this later
+	else if (StrEqual(sClassname, "obj_dispenser"))
```

Comment:
I should've mentioned no newlines between `}` to `else if`.
Also may be better to check `TFObjectType` instead of entity classname for teleporter and dispenser?